app/src/androidTest/java/net/skyscanner/backpack/BpkTestRunner.kt

```diff
@@ -17,15 +17,40 @@
  */
 
 package net.skyscanner.backpack
+
+import android.app.Activity
+import android.app.Application
+import android.content.Context
+import android.content.Intent
 import android.os.Bundle
 import androidx.test.runner.AndroidJUnitRunner
 import com.facebook.testing.screenshot.ScreenshotRunner
 
 class BpkTestRunner : AndroidJUnitRunner() {
+
   override fun onCreate(args: Bundle?) {
     ScreenshotRunner.onCreate(this, args)
     super.onCreate(args)
   }
+
+  override fun newApplication(cl: ClassLoader?, className: String?, context: Context?): Application =
+    super.newApplication(cl, className, context).apply {
+      registerActivityLifecycleCallbacks(object : Application.ActivityLifecycleCallbacks {
+        override fun onActivityPaused(activity: Activity) {}
+        override fun onActivityResumed(activity: Activity) {}
+        override fun onActivityStarted(activity: Activity) {}
+        override fun onActivityDestroyed(activity: Activity) {}
+        override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
+        override fun onActivityStopped(activity: Activity) {}
+        override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
+          BpkTestVariant.current.applyToActivity(activity)
+        }
+      })
+    }
+
+  override fun newActivity(cl: ClassLoader?, className: String?, intent: Intent?): Activity =
```

Comment:
We want to inject custom resources before anyone called getResources() -- otherwise there'll be an exception